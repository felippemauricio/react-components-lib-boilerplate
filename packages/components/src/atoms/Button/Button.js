import { string } from 'prop-types';
import { styleModifierPropTypes } from 'styled-components-modifiers';
import ButtonModifiers, { defaultModifier } from './Button.modifiers';
import { Button } from './Button.style';


Button.propTypes = {
  children: string.isRequired,
  modifiers: styleModifierPropTypes(ButtonModifiers),
};

Button.defaultProps = {
  modifiers: defaultModifier,
};


export default Button;
