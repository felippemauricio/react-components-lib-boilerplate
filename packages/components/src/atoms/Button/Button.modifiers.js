import { palette } from 'styled-theme';


const ButtonModifiers = {
  default: (props) => `
    background-color: ${palette('default', 0)(props)};
    color: ${palette('default', 1)(props)};
  `,
  primary: (props) => `
    background-color: ${palette('primary', 0)(props)};
    color: ${palette('primary', 1)(props)};
  `,
};

export const defaultModifier = 'default';


export default ButtonModifiers;
