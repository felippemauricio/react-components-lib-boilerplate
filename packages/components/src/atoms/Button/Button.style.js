import styled from 'styled-components';
import { applyStyleModifiers } from 'styled-components-modifiers';
import ButtonModifiers from './Button.modifiers';


const Button = styled.button`
  border: 2px solid palevioletred;
  border-radius: 3px;
  font-size: 1em;
  margin: 1em;
  padding: 0.25em 1em;

  ${applyStyleModifiers(ButtonModifiers)}
`;


export {
  Button,
};
