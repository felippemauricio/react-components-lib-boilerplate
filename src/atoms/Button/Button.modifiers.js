
const ButtonModifiers = {
  default: ({ theme }) => `
    background-color: palevioletred;
    color: white;
  `,
  primary: ({ theme }) => `
    background-color: white;
    color: palevioletred;
  `,
};

export const defaultModifier = 'default';


export default ButtonModifiers;
