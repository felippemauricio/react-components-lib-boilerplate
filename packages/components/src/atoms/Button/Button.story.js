import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import Button from './Button';


storiesOf(`Button`, module)
  .add('Button Default', () => {
    const onClick = action('Clicked');
    return (
      <Button onClick={onClick}>
        Default Button
      </Button>
    );
  })
  .add('Button Primary', () => {
    const onClick = action('Clicked');
    return (
      <Button onClick={onClick} modifiers="primary">
        Button Primary
      </Button>
    );
  });
