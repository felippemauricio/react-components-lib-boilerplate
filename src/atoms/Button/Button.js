import React from 'react';
import { string } from 'prop-types';
import { styleModifierPropTypes } from 'styled-components-modifiers';
import ButtonModifiers, { defaultModifier } from './Button.modifiers'; 
import { Button as Btn } from './Button.style';


function Button({ children, ...props }) {
  return (
    <Btn {...props}>
      {children}
    </Btn>
  );  
}

Button.propTypes = {
  children: string.isRequired,
  modifiers: styleModifierPropTypes(ButtonModifiers),
};

Button.defaultProps = {
  modifiers: defaultModifier,
};


export default Button;
