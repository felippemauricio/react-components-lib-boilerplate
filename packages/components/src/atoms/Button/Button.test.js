import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { ThemeProvider } from 'styled-components';
import Themes from '@react-components-lib-boilerplate/themes';
import Button from './Button';


describe('Atoms/Button', () => {
  it('Render default Button', () => {
    const tree = renderer.create(
      <ThemeProvider theme={Themes.main}>
        <Button>
          Default Button
        </Button>
      </ThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Render primary Button', () => {
    const tree = renderer.create(
      <ThemeProvider theme={Themes.main}>
        <Button modifiers="primary">
          Primary Button
        </Button>
      </ThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Button text are correct', () => {
    const buttonText = 'Hello World';
    const { root: instance } = renderer.create(
      <ThemeProvider theme={Themes.main}>
        <Button>
          {buttonText}
        </Button>
      </ThemeProvider>
    );
    const hasTextInsideButton = instance.find((el) => el.type === 'button'
      && el.children
      && el.children[0] === buttonText
    );
    expect(hasTextInsideButton).toBeTruthy();
  });
});
