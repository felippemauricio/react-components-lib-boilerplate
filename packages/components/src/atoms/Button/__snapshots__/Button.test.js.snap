// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/Button Render default Button 1`] = `
.c0 {
  border: 2px solid palevioletred;
  border-radius: 3px;
  font-size: 1em;
  margin: 1em;
  padding: 0.25em 1em;
  background-color: palevioletred;
  color: white;
}

<button
  className="c0"
>
  Default Button
</button>
`;

exports[`Atoms/Button Render primary Button 1`] = `
.c0 {
  border: 2px solid palevioletred;
  border-radius: 3px;
  font-size: 1em;
  margin: 1em;
  padding: 0.25em 1em;
  background-color: white;
  color: palevioletred;
}

<button
  className="c0"
>
  Primary Button
</button>
`;
